dm/syncer/syncer.go

```diff
@@ -3447,14 +3449,18 @@ func (s *Syncer) handleEventError(err error, startLocation, endLocation binlog.L
 	return terror.Annotatef(err, "startLocation: [%s], endLocation: [%s]", startLocation, endLocation)
 }
 
+var (
+	emptyMySQLGTIDSet, _ = gmysql.ParseMysqlGTIDSet("")
+)
+
 func (s *Syncer) adjustGlobalPointGTID(tctx *tcontext.Context) (bool, error) {
 	location := s.checkpoint.GlobalPoint()
 	// situations that don't need to adjust
 	// 1. GTID is not enabled
 	// 2. location already has GTID position
 	// 3. location is totally new, has no position info
 	// 4. location is too early thus not a COMMIT location, which happens when it's reset by other logic
-	if !s.cfg.EnableGTID || location.GTIDSetStr() != "" || location.Position.Name == "" || location.Position.Pos == 4 {
+	if !s.cfg.EnableGTID || location.GetGTID() != emptyMySQLGTIDSet || location.Position.Name == "" || location.Position.Pos == 4 {
```

Comment:
this will fail if upstream is a MariaDB

please create a public function inside `dm/pkg/binlog/` and use that

also, the other `location.GetGTIDStr() == ""` was left unchanged